src/test-exec.c

```diff
@@ -867,10 +871,20 @@ static void test_send_next_command(struct test_exec_context *ctx,
 		(void)imap_client_append_full(client, NULL, NULL, NULL,
 					      test_cmd_callback, &cmd);
 	} else if (strncasecmp(cmdline, "append ", 7) == 0 &&
-		   !append_has_body(ctx, cmdline+7, cmdline_len-7)) {
+		   !append_has_body(ctx, FALSE, cmdline+7, cmdline_len-7)) {
 		client->client.state = STATE_APPEND;
 		(void)imap_client_append(client, cmdline + 7, FALSE,
 					 test_cmd_callback, &cmd);
+	} else if (strncasecmp(cmdline, "replace ", 8) == 0 &&
+		   !append_has_body(ctx, TRUE, cmdline+8, cmdline_len-8)) {
+		client->client.state = STATE_APPEND;
+		(void)imap_client_replace(client, FALSE, cmdline + 8,
+					  test_cmd_callback, &cmd);
+	} else if (strncasecmp(cmdline, "uid replace ", 12) == 0 &&
+		   !append_has_body(ctx, TRUE, cmdline+12, cmdline_len-12)) {
```

Comment:
Well, would be nice to convert these to str_begins_icase_with()..

Reply:
changed